uamqp/aio/_connection_async.py

```diff
@@ -333,13 +384,26 @@ async def _process_incoming_frame(self, channel, frame):
             return True  #TODO: channel error
 
     async def _process_outgoing_frame(self, channel, frame):
+        if self.network_trace:
+            _LOGGER.info("-> %r", frame, extra=self.network_trace_params)
         if not self.allow_pipelined_open and self.state in [ConnectionState.OPEN_PIPE, ConnectionState.OPEN_SENT]:
             raise ValueError("Connection not configured to allow pipeline send.")
         if self.state not in [ConnectionState.OPEN_PIPE, ConnectionState.OPEN_SENT, ConnectionState.OPENED]:
             raise ValueError("Connection not open.")
+        now = time.time()
+        if (await self._get_local_timeout(now)) or (await self._get_remote_timeout(now)):
+            await self.close(
+                # TODO: check error condition
+                error=AMQPError(
+                    condition=ErrorCondition.ConnectionCloseForced,
+                    description="No frame received for the idle timeout."
+                ),
+                wait=False
+            )
+            return
         await self._send_frame(channel, frame)
 
-    def _get_local_timeout(self, now):
+    async def _get_local_timeout(self, now):
```

Comment:
Why should this be async?

Reply:
nah it doesn't have to -- I was too lazy :P

I have made this a helper method in `connection.py` such that it could be reused in async connection